go/types/opcache.go

```diff
@@ -8,15 +8,65 @@ import (
 	"encoding/binary"
 	"io/ioutil"
 	"os"
+	"sync/atomic"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/hash"
 	"github.com/syndtr/goleveldb/leveldb"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
 )
 
-func newOpCache(vrw ValueReadWriter) *opCache {
+const uint32Size = 4
+
+type OpCacheVender interface {
+	OpCache() OpCache
+}
+
+type OpCacheStore interface {
+	OpCacheVender
+	Destroy() error
+}
+
+type OpCacheVenderReadWriter interface {
+	OpCacheVender
+	ValueWriter
+	ValueReader
+}
+
+type OpCache interface {
+	MapSet(mapKey Value, mapVal Value)
+	SetInsert(val Value)
+	NewIterator() OpCacheIterator
+}
+
+type OpCacheIterator interface {
+	MapOp() sequenceItem
+	SetOp() sequenceItem
+	Next() bool
+	Release()
+}
+
+type LdbOpCacheStore struct {
+	ldb          *leveldb.DB
+	dbDir        string
+	collectionId uint32
+	vrw          ValueReadWriter
+}
+
+type LdbOpCache struct {
+	vrw   ValueReadWriter
+	colId uint32
+	store *LdbOpCacheStore
```

Comment:
I _think_ the only thing you do with `store` in `LdbOpCache` methods is access the `ldb` field. Why not just make this be `ldb *leveldb.DB`?